util/network-devp2p/src/discovery.rs

```diff
@@ -191,7 +183,7 @@ pub struct TableUpdates {
 }
 
 impl<'a> Discovery<'a> {
-	pub fn new(key: &KeyPair, public: NodeEndpoint, ip_filter: IpFilter) -> Discovery<'static> {
+	pub fn new(key: &KeyPair, public: NodeEndpoint, ip_filter: IpFilter) -> Discovery<'a> {
```

Comment:
it seems the previous `'static` lifetime was valid, why the change?

Reply:
We can removed the lifetime altogether by specifying `request_backoff: &'static [Duration]` in the struct

Reply:
> it seems the previous 'static lifetime was valid, why the change?

I thought the `static lifetime` for `request_backoff: &'a [Duration]` unnecessarily big? That's why, i.e, needs only `live` as long the `"a` AFAIU.

It seems like it's never changed after construction, so could could be `&'static Duration` I guess.

Any strong opinions?

Reply:
I think the reason the struct has a lifetime in the first place is to be able to pass any reference for `request_backoff`, but in practice it's always a (non-mut) reference to a static var, so I think we should remove the lifetime altogether.